### PR TITLE
57 bump up to sb 3.0

### PR DESCRIPTION
**Pull request by arnosthavelka**

### CODE REVIEW

File: sat-elk/src/main/java/com/github/aha/sat/elk/city/CityService.java [hunk 1]
```diff
@@ -60,6 +61,7 @@ List<City> parseFile(String csvFileName) {
 		try {
 			var csvFile = Path.of(csvFileName);
 			return csvMapper
+					.disable(FAIL_ON_MISSING_HEADER_COLUMNS)
```
Comment (arnosthavelka) [inline sat-elk/src/main/java/com/github/aha/sat/elk/city/CityService.java:63]: due to:
```
Caused by: com.fasterxml.jackson.dataformat.csv.CsvReadException: Missing 1 header column: ["id"]
 at [Source: (File); line: 1, column: 25]
	at com.fasterxml.jackson.dataformat.csv.CsvReadException.from(CsvReadException.java:25)
	at com.fasterxml.jackson.dataformat.csv.CsvParser._reportCsvMappingError(CsvParser.java:1356)
	at com.fasterxml.jackson.dataformat.csv.CsvParser._readHeaderLine(CsvParser.java:870)
	at com.fasterxml.jackson.dataformat.csv.CsvParser._handleStartDoc(CsvParser.java:889)
	at com.fasterxml.jackson.dataformat.csv.CsvParser.nextToken(CsvParser.java:680)
	at com.fasterxml.jackson.databind.ObjectReader._bindAndReadValues(ObjectReader.java:2206)
	at com.fasterxml.jackson.databind.ObjectReader.readValues(ObjectReader.java:1979)
	at com.github.aha.sat.elk.city.CityService.parseFile(CityService.java:65)
```

File: sat-jpa/pom.xml [hunk 2]
```diff
@@ -31,21 +31,35 @@
...
 			<groupId>org.hibernate</groupId>
 			<artifactId>hibernate-jpamodelgen</artifactId>
+			<version>6.1.6.Final</version>
+<!-- 			<version>${hibernate.version}</version> -->
 		</dependency>
 		<dependency>
 			<groupId>com.querydsl</groupId>
 			<artifactId>querydsl-jpa</artifactId>
+			<version>${querydsl.version}</version>
+			<classifier>jakarta</classifier>
```
Comment (arnosthavelka) [inline sat-jpa/pom.xml:40]: See https://github.com/querydsl/querydsl/issues/3296

File: sat-jpa/src/test/java/com/github/aha/sat/jpa/country/CountryRepositoryCustomTests.java [hunk 3]
```diff
@@ -40,13 +40,14 @@ void exactValues() {
 
 		@Test
 		void wildcard() {
-			var result = countryRepository.findAllCountriesBy("%an%", "%i%");
+			var result = countryRepository.findAllCountriesHavingCity("%an%", "%i%");
 
 			assertThat(result)
-					.hasSize(2)
-					.allSatisfy(c -> {
+					.singleElement()
```
Comment (arnosthavelka) [inline sat-jpa/src/test/java/com/github/aha/sat/jpa/country/CountryRepositoryCustomTests.java:45]: SB 3.0 -> 1 record
```
select c2_0.id,c2_0.name
from city c1_0 
	join country c2_0 on c2_0.id=c1_0.country_id 
where c1_0.name like ? escape '!' and c1_0.state like ? escape '!'
```

SB 2.7 -> 2 record
```
select country1_.id as id1_1_, country1_.name as name2_1_ 
from city city0_ 
	inner join country country1_ on city0_.country_id=country1_.id 
where (city0_.name like ? escape '!') and (city0_.state like ? escape '!')
```

File: sat-jpa/src/main/java/com/github/aha/sat/jpa/JpaApplication.java [hunk 4]
```diff
@@ -13,15 +10,4 @@ public static void main(String[] args) {
         SpringApplication.run(JpaApplication.class, args);
     }
 
-	/**
-	 * Usage: http://localhost:8080/console/ (JDBC URL=jdbc:h2:mem:testdb)
-	 * 
-	 * @see https://dzone.com/articles/using-the-h2-database-console-in-spring-boot-with
-	 */
-	@Bean
-	ServletRegistrationBean<WebServlet> h2servletRegistration() {
```
Comment (arnosthavelka) [inline sat-jpa/src/main/java/com/github/aha/sat/jpa/JpaApplication.java:22]: not needed now as it's auto-configured automatically with H2

**Comment by maddelar** [comment]
how to generate the Q classes with jdk17 and springboot 3.1.1? could you please provide some documentation?

**Comment by arnosthavelka** [comment]
> how to generate the Q classes with jdk17 and springboot 3.1.1? could you please provide some documentation?

@maddelar what do you mean? My develop branch is already using JDK17 & SpringBoot 3.1.2 and I already use Q classes there. See e.g. `CountryDzoneTests`

**Pull request merged**